### PR TITLE
Mark optimized abs as being nogil-safe.

### LINKED ISSUE #2746: Optimized abs should be nogil-safe
From the following:
```cython
cdef double foo(double x):
    return abs(x)

cdef int bar(int x):
    return abs(x)
```
the resulting C/C++ code calls `fabs`/`abs` thanks to #1255. If you add `nogil` on these functions, then Cython will complain with:
```
    return abs(x)
             ^
------------------------------------------------------------
test_abs.pyx:2:14: Calling gil-requiring function not allowed without gil
```
Because of #1837, I'm not sure about `bar`, but at least `foo` should not require the GIL.

### PR DESCRIPTION
**Pull request by QuLogic**
I'm not entirely sure how to get the single test file to run, so I'm not sure if they're written correctlyy, but at least it compiles.

Fixes #2746.

### CODE REVIEW

**Comment by scoder** [comment]
Thanks. Looks like your changes broke the `int_abs()` and `long_abs()` tests. To run the `builtin_abs` tests separately, just pass that name to `runtests.py` as an argument, e.g.
`python3 runtests.py --debug -vv builtin_abs`

Regarding the implementation, I think it would be nicer if `nogil=True` was just a new keyword argument to `BuiltinFunction`, which means adding it as a constructor argument to the `Signature` class in `TypeSlots.py`, and build the appropriate `CFuncType` in `Signature.function_type()`.

File: Cython/Compiler/Builtin.py [hunk 1]
```diff
@@ -92,13 +92,33 @@ def declare_in_type(self, self_type):
...
+                        is_strict_signature=True, nogil=True)),
     BuiltinFunction('abs',        "l",    "l",     "labs",
-                    is_strict_signature = True),
+                    func_type=PyrexTypes.CFuncType(
+                        PyrexTypes.c_long_type, [
+                            PyrexTypes.CFuncTypeArg(
+                                "arg", PyrexTypes.c_long_type, None)
+                        ],
+                        is_strict_signature=True, nogil=True)),
     BuiltinFunction('abs',        None,    None,   "__Pyx_abs_longlong",
```
Comment (scoder) [inline Cython/Compiler/Builtin.py]: `nogil=True` also applies here.

Comment (QuLogic) [reply Cython/Compiler/Builtin.py]: It already has it; that's where I found the `CFuncType` setup originally.

**Comment by QuLogic** [comment]
> To run the `builtin_abs` tests separately, just pass that name to `runtests.py` as an argument, e.g.

Ah, I tried passing the filename instead.

**Comment by scoder** [comment]
Perfect! Thanks!

**Pull request merged**